README
#   $Id: README,v 1.9 2007-01-26 22:24:06 adam Exp $

XML/RSS/Tools version 0.20
==========================

RSS/RDF feeds are commonly available ways of distributing the latest news
about a given web site for news syndication. This module provides a VERY
high level way of manipulating them. You can easily use LWP, the XML::RSS
and XML::LibXSLT do to this yourself.

STATUS

This module is BETA code, it may have undocumented features "bugs", and
the API may change. It is currently under development and so now is the
time to add features to the code base before the API becomes fixed.

INSTALLATION

To install this module type the following:

   perl Makefile.PL
   make
   make test
   make install

It will install okay on Windows with ActiveState Perl without a compiler
just use the free Microsoft nmake tool instead of make.

During the install it will try to connect to the Internet to test the HTTP
toolchain.

DEPENDENCIES

This module requires these other non-core modules for full functionality:

  XML::LibXSLT XML::LibXML
  XML::RSS
  URI
  LWP or HTTP::GHTTP or HTTP::Lite

XML::RSS older versions than version 0.98 have a number of defects and
limitations, please upgrade if possible.

XML::LibXML and XML::LibXSLT rely on the GNOME libxml2 and libxslt libraries
which need to be installed before attempting to install the Perl modules.
XML Catalog support in certain versions of XML::LibXML appears to be broken
if you are unable to use Catalogs please let me know so that I can better
document the module.

Please be aware that the API to the URI module changed, at version 1.32,
and hence how you use the file scheme changed, see the URI changelog for
more details.

OLDER VERSIONS

To keep CPAN trim, older versions of this module will be continually culled as
newer versions are loaded. Older versions are archive on backPAN here:
http://history.perl.org/backpan/authors/id/A/AT/ATRICKETT/  and I keep copies
myself which are also available upon request.

COPYRIGHT AND LICENCE

XML::RSS::Tools, Copyright (C) iredale Consulting 2002-2007

See POD for full details.


CHANGES
#   $Id: CHANGES,v 1.15 2007-01-26 22:24:06 adam Exp $

Revision history for Perl extension XML::RSS::Tools.

0.20  Xxx xx Month 2007
	- Wash code through PBP/Perltidy
    - Remove article, provide link instead
    - Remove example rss file
    - Convert documented files to example files
    - Fix META.yml bug
    - More tests and better coverage
    - Using MAKEFILE.SKIP

0.16  Fri 15 January 2006
	- Fix bug in xsl import - thanks to Martin Flack
	- Support Build.PL
	- Added extra POD tests
	- More additions to the POD

0.15  Sun 06 February 2005
	- Catalog support in some versions of XML::LibXML appears to be broken
	- Minor test changes and fixes to POD and build process
	- URI has changed, tests changes to adapt
	- Module code unchanged

0.14  Sat 24 April 2004
	- Configuration diagnostics in tests
	- Minor doc corrections/changes
	- META.yml file corrections and additions
	- Module code unchanged

0.13  Sun 28 March 2004
	- Implement HTTP Proxy support - no tests yet
	- Supplementary documentation converted from HTML to POD
	- Minor code and test additions/changes
	- Layout of files revised, initial attempt to make package Module::Build compatible - still not complete

0.12  Thu 12 June 2003
	- Lots of fixes to the build & test process
		* Example 5 added to the manifest
		* http-tests.t defect corrected
		* Makefile corrected
	- Module code unchanged

0.11  Wed 11 June 2003
	- POD Improvements
	- Corrections to included RSS feed
	- Example 5 added
	- Documentation improvements, print style-sheet added for easier printing
	- Fix & changes to Makefile.PL, should auto detect a HTTP server for testing purposes
	- Tests overhauled and cleaned up
	- Defect in xsl_uri subroutine detected and corrected

0.10  Sat 04 May 2003
	- Initial XML Catalog support, see www.oasis-open.org
	- Added ability to select which HTTP client to use
	- Added support for HTTP::Lite as a HTTP client
	- Syntax tests on constructor added
	- More tests
	- Added an RSS feed of the documentation

0.09  Fri 28 March 2003
	- Tested with latest XML::RSS Versions 0.98 and 1.02
	- Changes and additions to documentation and HTML
	- Warning message in Makefile.PL removed now that XML::RSS has moved out of beta
	- Bump versions of dependents up in Makefile.PL
	- Experimental support for FileHandle objects, internally and from an external call
	- More tests, including a POD test

0.08  Fri 10 January 2003
	- Incorporates "The Perl Review" Article - Processing RSS Files with XSLT
	- Minor changes to HTML, and updated links  
	- Removed Diagnostics pragma
	- Warning message added to Makefile.PL
	- Extra tests for RSS versions
	- Ability to disable RSS normalisation
	- Extra code examples

0.07  Sun 24 November 2002
	- More minor changes
	- Tests re-organised, re-ordered, and expanded
	- Additional Documentation
	- Uses the URI module for parsing URIs
	- File can be passed by the URI handler

0.06  Thu 01 August 2002
	- Cosmetic changes to the HTML, and minor additions
	- Test script corrections
	- Module code unchanged

0.05  Wed 31 July 2002
	- Null URIs dealt with
	- Extra example script provided
	- Minor changes
	- Tests changed for extra CPANPLUS compatibility

0.04  Mon 13 May 2002
	- Minor changes
	- Defect in test script corrected

0.03  Sun 12 May 2002
	- More minor changes and fixes
	- Example script included
	- Uploaded to more sensible location on CPAN

0.02  Sat 11 May 2002
	- Test suite created
	- Uploaded to CPAN as alpha code

0.01  Sun 28 Apr 2002
	- original version; created by h2xs 1.21 with options
		-X -n XML::RSS::Tools


Build.PL
#	$Id: Build.PL,v 1.3 2007-01-26 22:24:06 adam Exp $

use Module::Build;

my @tests  = qw(t/config.t t/use.t t/basic-tests.t t/standard-tests.t t/version-tests.t t/filehandle-tests.t t/error-tests.t);
my $ghttp;
my $lite;

print "Checking which extra POD testing modules you have installed...\n";

eval { require Test::Pod; };
push @tests, 't/pod-test.t' unless ($@);

eval { require Pod::Coverage; };
push @tests, 't/pod-cover-1.t' unless ($@);

eval { require Test::Pod::Coverage; };
push @tests, 't/pod-cover-2.t' unless ($@);

eval { require IO::Capture; };
if (! $@) {
    push @tests, 't/debug-tests.t';
    push @tests, 't/no-debug-tests.t';
};

print "Checking which HTTP clients you have installed...\n";

eval { require HTTP::GHTTP; };
if ($@) {
	print "You don't have HTTP::GHTTP installed.\nIt's not a requirment, but it's one\nof the fastest HTTP clients about.\n";
} else {
	print "Good you have HTTP::GHTTP installed.\n";
	$ghttp++;
}

eval { require HTTP::Lite; };
if ($@) {
	print "\nYou don't have HTTP::Lite installed either.\nIt's also not a requirment, but it's faster\nthan LWP.\n" unless $ghttp;
} else {
	$lite++;
	if ($ghttp) {
		print "You have HTTP::Lite installed too.\n";
	} else {
		print "\nBut you do have HTTP::Lite installed.\n";
	}
}

eval { require LWP };
if ($@) {
	print "\nYou don't have LWP installed!\n" unless $ghttp && $lite;
} else {
	if ($ghttp || $lite) {
		print "You have LWP installed too.\n";
	} else {
		print "\nBut at least you have LWP installed.\n";
	}
}

eval { require XML::LibXML };
if ($@) {
	print "\nYou don't have XML::LibXML installed!\n";
} else {
	if ($XML::LibXML::VERSION < 1.53) {
		print "Your version of LibXML is too old to support XML Catalog.\n";
		push @tests, "t/no-catalog.t";
	} else {
		push @tests, "t/catalog-tests.t";
	}
}

# Check if we have internet connection
# (I lifted this code from Net::DNS which lifted it from LWP...)
print "\n\nProbing for a HTTP Server, this may take a few seconds...\n\n";
require IO::Socket;
my $hostname = hostname;
my $s = IO::Socket::INET->new(
	PeerAddr => "www.iredale.net:80",
	Timeout  => 10
);

if ($s) {
	close($s);
	print "Good, you are connected to the Internet!\nWill test HTTP clients for you...\n";
	push @tests, "t/http-tests.t";
} elsif ($s = IO::Socket::INET->new(PeerAddr => "$hostname:80", Timeout  => 10)) {
		print "Good, $hostname is running a HTTP server.\nWill test HTTP clients for you...\n";
		push @tests, "t/http-tests.t";
		close($s);
} else {
	print "Can't reach the Internet or find a HTTP server on $hostname\nAborting HTTP client tests.\n";
}

my $build = Module::Build->new(
	module_name => 'XML::RSS::Tools',
	license     => 'gpl',
	test_files  => \@tests,
	requires    => {
		'perl'  => '5.6.1',
		'URI'	=> '1.31',
        'LWP'   => '5.7',
        'XML::LibXSLT' => '1.5',
        'XML::LibXML'  => '1.5',
        'XML::RSS'     => '1.0',
	},
);

$build->create_build_script;
